test/test_tree_2d_mhd.jl

```diff
@@ -148,6 +148,36 @@ end
     end
 end
 
+@trixi_testset "elixir_mhd_ec_float32.jl" begin
+    @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_mhd_ec_float32.jl"),
+                        l2=Float32[0.036355644,
+                                   0.042947736,
+                                   0.04294775,
+                                   0.025748005,
+                                   0.16211236,
+                                   0.017452478,
+                                   0.017452475,
+                                   0.026877576,
+                                   2.0951437f-7],
+                        linf=Float32[0.22100884,
+                                     0.28798944,
+                                     0.2879896,
+                                     0.20858234,
+                                     0.8812654,
+                                     0.09208202,
+                                     0.09208143,
+                                     0.14795381,
+                                     2.0912607f-6])
+    # Ensure that we do not have excessive memory allocations
+    # (e.g., from type instabilities)
+    let
+        t = sol.t[end]
+        u_ode = sol.u[end]
+        du_ode = similar(u_ode)
+        @test (@allocated Trixi.rhs!(du_ode, u_ode, semi, t)) < 1000
+    end
+end
```

Comment:
Did you check whether this uses only `Float32` computations using something like LIKWID?

Reply:
No, have you or @huiyuxie experience with this?

Reply:
See for example https://github.com/trixi-framework/Trixi.jl/pull/2042

Reply:
I can do it all together later if you can wait. I only have access to Windows/macOS, so I would need to request a cloud Linux server and it takes some time. If any of you already have a Linux server, it would be better for you to do this first.

Reply:
But I think it would be better if you can do it @ranocha

Reply:
Hm, unfortunately my processor is not supported by LIKWID, neither the processor of our local compute node. 

With the poor mans-solution 

```julia
function Base.convert(::Type{Float64}, x::Float32)
    error("convert Float32 to Float64")
    return Float64(x)
end
```
I was able to check that `rhs!` has no conversions.

Interestingly, the trigonometric functions can do some crazy stuff, see

https://github.com/JuliaLang/julia/blob/80a279152bd090cca0a3bed598f0059cffa74b26/base/special/rem_pio2.jl#L282-L283

Reply:
It seems unavoidable but I think it is okay to have a small amount of double-precision computation.

Reply:
Yeah, but that would be replaced on GPU kernels. Thanks!